firefox_puppeteer/api/prefs.py

```diff
@@ -202,3 +202,13 @@ def set_pref(self, pref_name, value):
             """, script_args=[pref_name, value])
 
         assert retval
+
+    def persist_prefs(self):
+        """Persist current pref values to the default prefs file (typically prefs.js)
+        """
+        with self.marionette.using_context('chrome'):
+            self.marionette.execute_script("""
+              Cu = Components.utils;
+              Cu.import("resource://gre/modules/Services.jsm");
+              Services.prefs.savePrefFile(null);
```

Comment:
My hypothesis is that marionette kills the browser in a way that doesn't hit the pref persisting code. I'll try to figure out what's going on.